src/Compilers/CSharp/Portable/Symbols/Metadata/PE/NativeIntegerTypeDecoder.cs

```diff
@@ -104,7 +104,9 @@ private NamedTypeSymbol TransformNamedType(NamedTypeSymbol type)
                         {
                             throw new UnsupportedSignatureContent();
                         }
-                        return _transformFlags[_index++] ? type.AsNativeInteger() : type;
+                        return _transformFlags[_index++] ?
+                            type.AsNativeInteger() :
+                            (type.NativeIntegerUnderlyingType ?? type);
```

Comment:
Is it correct that this change is so that we can handle all four cases: type is nint or IntPtr, and transform flag is set or not?
If so, I don't understand why we would get a `nint` type. I would expect only `IntPtr` here.

Reply:
Yes, all four cases are possible. Updated.

---
In reply to: [485230638](https://github.com/dotnet/roslyn/pull/47536#discussion_r485230638) [](ancestors = 485230638)